Makefile.pre.in

```diff
@@ -587,7 +587,7 @@ clinic: check-clean-src $(srcdir)/Modules/_blake2/blake2s_impl.c
 
 # Build the interpreter
 $(BUILDPYTHON):	Programs/python.o $(LIBRARY_DEPS)
-	$(LINKCC) $(PY_CORE_LDFLAGS) $(LINKFORSHARED) -o $@ Programs/python.o $(BLDLIBRARY) $(LIBS) $(MODLIBS) $(SYSLIBS)
+	$(LINKCC) $(PY_CORE_LDFLAGS) $(LINKFORSHARED) -o $@ Programs/python.o -Wl,--whole-archive $(BLDLIBRARY) -Wl,--no-whole-archive $(LIBS) $(MODLIBS) $(SYSLIBS)
```

Comment:
Why do we need this line? It's directly breaking macOS compilation.

Reply:
The option is of course required but the error is this linker option does not exist in clang which macOS uses. Is there a way to bypass this issue?

Reply:
We should check if the issue is reproducible in clang, but seems that most cases will need a bunch of conditionals to support different compilers.

Reply:
Also we've compiled cpython for a long time without this option. So is it worth adding those options now?

Reply:
> Also we've compiled cpython for a long time without this option. So is it worth adding those options now?

Yes because is fixing a bug 😉

Notice also that this bug can bite us in worst ways, we have been lucky that this only happened for this symbol.

Reply:
Ok. But what should be the workaround? This option is only applicable to gcc or g++.

Reply:
> Ok. But what should be the workaround? This option is only applicable to gcc or g++.

We need to find the same option in other compilers.

Reply:
@vstinner what (probably) happens here is that this Symbol as no user during linkage, so it gets removed. If you want it in, you can force a reference from a *live* function (e.g. taking its address and storing it etc).

An alternative to passing ``--whole-archive`` would be to use the ``--require-defined=symbol`` linker option.

Reply:
> An alternative to passing --whole-archive would be to use the --require-defined=symbol linker option.

Python exports 1611 symbols. Should I specify this option 1611 times?

Reply:
I am marking this thread as resolved.